lib/internal/quic/quic.js

```diff
@@ -0,0 +1,2694 @@
+
+  /** @type {bigint} */
+  get id() { return this[kState].id; }
+}
+ObjectDefineProperties(Stream.prototype, {
+  stats: { __proto__: null, enumerable: true },
+  state: { __proto__: null, enumerable: true },
+  session: { __proto__: null, enumerable: true },
+  id: { __proto__: null, enumerable: true },
+});
+
+class BidirectionalStream extends Stream {
+  constructor() {
+    throw new ERR_ILLEGAL_CONSTRUCTOR('BidirectionalStream');
+  }
+
+  /** @type {ReadableStream} */
+  get readable() { return this[kReadable]; }
+
+  /** @type {WritableStream} */
+  get writable() { return this[kWritable]; }
+
+  [kInspect](depth, options) {
+    if (depth < 0)
+      return this;
+
+    const opts = {
+      ...options,
+      depth: options.depth == null ? null : options.depth - 1,
+    };
+
+    return `BidirectionalStream ${inspect({
+      stats: this.stats,
+      state: this.state,
+      session: this.session,
+      readable: this.readable,
+      writable: this.writable,
+    }, opts)}`;
+  }
+}
+ObjectDefineProperties(BidirectionalStream.prototype, {
+  readable: { __proto__: null, enumerable: true },
+  writable: { __proto__: null, enumerable: true },
+});
+
+class UnidirectionalInboundStream extends Stream {
+  constructor() {
+    throw new ERR_ILLEGAL_CONSTRUCTOR('UnidirectionalInboundStream');
+  }
+
+  /** @type {ReadableStream} */
+  get readable() { return this[kReadable]; }
+
+  [kInspect](depth, options) {
+    if (depth < 0)
+      return this;
+
+    const opts = {
+      ...options,
+      depth: options.depth == null ? null : options.depth - 1,
+    };
+
+    return `UnidirectionalInboundStream ${inspect({
+      stats: this.stats,
+      state: this.state,
+      session: this.session,
+      readable: this.readable,
+    }, opts)}`;
+  }
+}
+ObjectDefineProperties(UnidirectionalInboundStream.prototype, {
+  readable: { __proto__: null, enumerable: true },
+});
+
+class UnidirectionalOutboundStream extends Stream {
+  constructor() {
+    throw new ERR_ILLEGAL_CONSTRUCTOR('UnidirectionalOutboundStream');
+  }
+
+  /** @type {WritableStream} */
+  get writable() { return this[kWritable]; }
+
+  [kInspect](depth, options) {
+    if (depth < 0)
+      return this;
+
+    const opts = {
+      ...options,
+      depth: options.depth == null ? null : options.depth - 1,
+    };
+
+    return `UnidirectionalOutboundStream ${inspect({
+      stats: this.stats,
+      state: this.state,
+      session: this.session,
+      writable: this.writable,
+    }, opts)}`;
+  }
+}
+ObjectDefineProperties(UnidirectionalOutboundStream.prototype, {
+  writable: { __proto__: null, enumerable: true },
+});
+
+/**
+ * @param {object} handle
+ * @returns {BidirectionalStream}
+ */
+function createBidirectionalStream(handle, session) {
+  const instance = ReflectConstruct(EventTarget, [], BidirectionalStream);
+  instance[kHandle] = handle;
+  instance[kStats] = createStreamStats(handle.stats);
+  instance[kState] = createStreamState(handle.state);
```

Comment:
Makes me a bit nervous that these names are a single character apart. High chance of typos in future maintenance.

Reply:
Not ignoring this comment, just plan to address it in a subsequent iteration